diag/diag.proto

```diff
@@ -0,0 +1,236 @@
+//
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// This file defines the gNOI APIs used to perform diagnostic operations on a
+// switching box.
+syntax = "proto3";
+
+package gnoi.diag;
+
+// The Diag service exports to main set of RPCs:
+// 1- BERT related RPCs: Used to perform Bit Error Rate Test (BERT)
+//    operations on a set of ports.
+// 2- BURNING related RPCs: Used to perform a vendor-provided Burnin test on the
+//    switch to ensure the box is ready to start serving traffic. Burnin tests
+//    are typically run by HwOps in the datacenter, as part of turnup or repair
+//    workflow.
+service Diag {
+  // Starts BERT operation on a set of ports. Each BERT operation is uniquely
+  // identified by an ID, which is given by the caller. The caller can then
+  // use this ID (as well as the list of the ports) to stop the BERT operation
+  // and/or get the BERT results. This RPC is expected to return an error status
+  // in the following situations:
+  // - When BERT operation is not supported on any of the ports specified by
+  //   the request.
+  // - When BERT is already in progress on any port specified by the request.
+  // - In case of any low-level HW/SW internal errors.
+  // The RPC returns an OK status of none of these situations is encountered.
+  rpc StartBERT(StartBERTRequest) returns(StartBERTResponse) {}
```

Comment:
Curious on your thoughts, but in another PR, I am thinking we should make sure there is an OpenConfig state leaf to describe that an interface is being used for testing. Since this is a service affecting activity, we would want a way for streaming telemetry to indicate that even though the interface may be UP/UP, it is currently under test.

Reply:
oh definitely agree with that. This was actually in our todo. Since BERT operation are long lived operations, the oper state of an interface MUST be able to capture this. 

Internally we actually do have a specific state for a port when it is in so called "diag mode". We can definitely extend gnmi to capture that as well.

Reply:
We actually have a `TESTING` state in `/interfaces/interface/state/oper-status` already. Would this be OK for this use case?

Reply:
For optical devices, I think we also need this in the terminal-device/logical channels. We can expand this enum possibly:
https://github.com/openconfig/public/blob/e8981a183157a5f730f3d054da63f9314f43efb4/release/models/optical-transport/openconfig-terminal-device.yang#L767

Reply:
Thanks, for our use case TESTING state should be enough I think.